### PR TITLE
[Feat]: add RSK

### PR DESCRIPTION
**Pull request by leablock**
# Name of the feature/issue

- Add RSK Mainnet and Testnet
- Add helper to add new a new network that the user already had (use its block eplorer url, description and order)
- 
## Description
- Add as default networks
  - Use nodes that support "eth_getLogs" https://developers.rsk.co/rif/identity/mvp/run/#1-run-the-infrastructure
  - Add the nodes in chains-assets https://github.com/block-wallet/chains-assets/pull/74 so that they're verified
- Add migration
- Add both in chain-fee https://github.com/block-wallet/chain-fee-data-service/pull/52
- Disable "showGasLevels" because "slow" level may be enough in some cases and the transaction could be rejected

image
image
image

### CODE REVIEW

File: packages/background/src/infrastructure/stores/migrator/migrations/migration-52.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+} from '../../../../controllers/blank-deposit/types';
+import { normalizeNetworksOrder } from '../../../../utils/networks';
+
+/**
+ * This migration adds RSK Mainnet and RSK Testnet
+ */
+export default {
+    migrate: async (persistedState: BlankAppState) => {
+        const { availableNetworks } = persistedState.NetworkController;
+        const updatedNetworks = { ...availableNetworks };
```
Comment (fpezzola) [inline packages/background/src/infrastructure/stores/migrator/migrations/migration-52.ts:20]: We have to take into account that the user may already have added this network to his wallet. So, we should first look for the RSK chain ID in the user's network list and if it exists:
- Delete the user's RSK network record, indexed as `CHAIN-30`
- Update some of our parameters with the user's configured network (like the RPC url, name, order, so on...)

Comment (fpezzola) [reply packages/background/src/infrastructure/stores/migrator/migrations/migration-52.ts:20]: You can use the `getNonNativeNetworkKey` helper to get the Key

**Review by fpezzola — CHANGES_REQUESTED** [review]
Please read my comment

File: packages/background/src/infrastructure/stores/migrator/migrations/migration-53.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ */
+export default {
+    migrate: async (persistedState: BlankAppState) => {
+        const { availableNetworks } = persistedState.NetworkController;
+        let updatedNetworks = { ...availableNetworks };
+        const networkController = new NetworkController(
+            persistedState.NetworkController
+        );
+
+        const rsk = {
```
Comment (fpezzola) [inline packages/background/src/infrastructure/stores/migrator/migrations/migration-53.ts:29]: Why do you duplicate this? Can't you use `INITIAL_NETWORKS.RSK`?

File: packages/background/src/infrastructure/stores/migrator/migrations/migration-53.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            blockExplorerUrls: ['https://explorer.rsk.co'],
+            actionsTimeIntervals: { ...ACTIONS_TIME_INTERVALS_DEFAULT_VALUES },
+            tornadoIntervals: {
+                depositConfirmations: DEFAULT_TORNADO_CONFIRMATION,
+                derivationsForward: DERIVATIONS_FORWARD,
+            },
+            nativelySupported: true,
+        };
+
+        updatedNetworks = addNetworkWithPreviousUserValues(
```
Comment (fpezzola) [inline packages/background/src/infrastructure/stores/migrator/migrations/migration-53.ts:60]: I'd limit the scope of the helper to only merge two configurations. 

```
//migration53
const userCurrentNetworkConfig = persistedState.NetworkController.find(network => network.chainId === INITIAL_NETWORKS.RSK.chainId || {};

updatedNetworks['RSK']= newNativelySupportedNetwork(INITIAL_NETWORKS.RSK, userCurrentNetworkConfig);


//networks.test
const newNativelySupportedNetwork(baseNetworkConfig: Network, userCurrentNetworkConfig: Network) : Network => {
  return { 
     ....baseNetworkConfig,
    order: userCurrentNetworkConfig.order ?? baseNetworkConfig.order,
    desc: userCurrentNetworkConfig?.desc || baseNetworkConfig.desc,
    rpcUrl: userCurrentNetworkConfig.rpcUrl || baseNetworkConfig.rpcUrl,
    blockExplorerUrl: userCurrentNetworkConfig.blockExplorerUrl || baseNetworkConfig.blockExplorerUrl,
    nativelySupported: true
   }
}
```

File: packages/background/src/infrastructure/stores/migrator/migrations/migration-53.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            'RSK_TESTNET',
+            rskTestnet,
+            updatedNetworks,
+            networkController
+        );
+
+        updatedNetworks.LOCALHOST = {
+            ...updatedNetworks.LOCALHOST,
+            order: updatedNetworks.LOCALHOST.order++,
```
Comment (fpezzola) [inline packages/background/src/infrastructure/stores/migrator/migrations/migration-53.ts:109]: `order: updatedNetworks.LOCALHOST.order + 1,`

**Review by fpezzola — APPROVED** [review]
lgtm! great work 👏

**Review by julianariel — APPROVED** [review]

**Pull request merged**